### PR TITLE
Blogging Reminders: Replace references to "goals" with "reminders"

### LINKED ISSUE #14895: Blogging Reminders: All references to "Goals" should be replaced with "Reminders"
Internal ref: p1623427096156100-slack-C01UWB9GD1Q

### PR DESCRIPTION
**Pull request by renanferrari**
Fixes #14895

This updates and renames various strings in order to replace all references to "goals" with "reminders". More specifically:

- Settings screen:
  - Blogging ~goals~ reminders
- First screen (Prologue):
  - Title: Set your blogging ~goals~ reminders
  - Body: Your post is publishing… in the meantime, set up your blogging ~goals to get reminders, and track your progress.~ reminders on days you want to post.
  - Button: Set ~goals~ reminders
- Last screen (Epilogue):
  - You can update this anytime via My Site > Settings > Blogging ~goals.~ reminders.

To test:

- Make sure you have built a debug variant of the app.
- Clear app data and log in.
- Make sure the feature flag is turned on.

Go through the entire Blogging Reminders flow and make sure no reference to "goal" exists.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -3636,12 +3632,14 @@
...
-    <string name="set_your_blogging_goals_title">Set your blogging goals</string>
-    <string name="set_your_blogging_goals_message">Your post is publishing… in the meantime, set up your blogging goals to get reminders, and track your progress.</string>
-    <string name="set_your_blogging_goals_button">Set goals</string>
-    <string name="blogging_goals_n_a_week">%s a week</string>
-    <string-array name="blogging_goals_count" translatable="false">
+    <!-- Blogging Reminders -->
+    <string name="blogging_reminders_notification_title">%s, it\'s time to blog</string>
+    <string name="blogging_reminders_notification_text">This is your reminder to blog today &#9997;</string>
+    <string name="set_your_blogging_reminders_title">Set your blogging reminders</string>
+    <string name="set_your_blogging_reminders_message">Your post is publishing… in the meantime, set up your blogging goals to get reminders, and track your progress.</string>
```
Comment (planarvoid) [inline WordPress/src/main/res/values/strings.xml]: I think this is on purpose but just to be sure, we still want the word "goals" in here, right?

Comment (renanferrari) [reply WordPress/src/main/res/values/strings.xml]: We actually wanted to update this as well but simply replacing "goals" with "reminders" wouldn't work here as the phrase would get redundant. 

I brought this up for discussion (internal ref: p1624445050255400-slack-C01UWB9GD1Q) and was waiting for a decision to be made before I could take this out of draft. 

I've updated the string on f81d853a and this is now ready for review.

**Review by ravishanker — APPROVED** [review]

**Comment by ravishanker** [comment]
LGTM 👍  
- just need to resolve conflicts and lint errors

**Review by planarvoid — APPROVED** [review]
LGTM 👍

**Pull request merged**